Linting errors in v0.2.7_dev

* IceNet version: v0.2.7_dev
* Python version: 3.8.18
* Operating System: Ubuntu 22.04.3 LTS

### Description

Running [ruff](https://github.com/astral-sh/ruff) linter shows pep8/PyFlakes errors/violation.

### What I Did

```bash
$ ruff check icenet/
icenet/data/cli.py:111:20: E721 Do not compare types, use `isinstance()`
icenet/data/dataset.py:148:16: E721 Do not compare types, use `isinstance()`
icenet/data/loaders/dask.py:140:31: E721 Do not compare types, use `isinstance()`
icenet/data/loaders/dask.py:143:28: E721 Do not compare types, use `isinstance()`
icenet/data/loaders/dask.py:469:16: E721 Do not compare types, use `isinstance()`
icenet/data/process.py:113:12: E721 Do not compare types, use `isinstance()`
icenet/data/processors/osi.py:30:16: E721 Do not compare types, use `isinstance()`
icenet/data/processors/utils.py:74:63: E712 Comparison to `True` should be `cond is True` or `if cond:`
icenet/data/processors/utils.py:79:71: E712 Comparison to `True` should be `cond is True` or `if cond:`
icenet/data/producers.py:257:24: E721 Do not compare types, use `isinstance()`
icenet/model/train.py:6:8: F401 [*] `pkg_resources` imported but unused
icenet/model/train.py:10:20: F401 [*] `pprint.pformat` imported but unused
icenet/plotting/data.py:17:22: F401 [*] `matplotlib` imported but unused
icenet/plotting/forecast.py:740:18: F541 [*] f-string without any placeholders
icenet/plotting/forecast.py:742:18: F541 [*] f-string without any placeholders
icenet/plotting/forecast.py:906:18: F541 [*] f-string without any placeholders
icenet/plotting/forecast.py:908:18: F541 [*] f-string without any placeholders
icenet/plotting/utils.py:435:26: F821 Undefined name `np`
icenet/plotting/utils.py:436:26: F821 Undefined name `np`
Found 19 errors.
[*] 7 fixable with the `--fix` option (2 hidden fixes can be enabled with the `--unsafe-fixes` option).
```

### Possible solutions

* ~~E721: Swap type(obj) checks for isinstance(obj, ...).~~
* E721: Add to ignore in checks.
* E712: False positive -> Add check ignore on lines, correct usage for array.
* F401: Remove `pkg_resources` import - unused in code, remove this specific `pprint.pformat` import, ignore `matplotlib` import line
* F541: Change from f-string to normal.
* F821: missing numpy import, bug?

--- a/icenet/data/processors/utils.py
+++ b/icenet/data/processors/utils.py
@@ -71,12 +71,12 @@ def sic_interpolate(da: object,
                 nan_neighbour_arr[-1, :] = False
 
             if np.sum(nan_neighbour_arr) == 1:
-                res = np.where(np.array(nan_neighbour_arr) == True)
+                res = np.where(np.array(nan_neighbour_arr) == True)  # noqa: E712
                 logging.warning("Not enough nans for interpolation, extending {}".format(res))
                 x_idx, y_idx = res[0][0], res[1][0]
                 nan_neighbour_arr[x_idx-1:x_idx+2, y_idx] = True
                 nan_neighbour_arr[x_idx, y_idx-1:y_idx+2] = True
-                logging.debug(np.where(np.array(nan_neighbour_arr) == True))
+                logging.debug(np.where(np.array(nan_neighbour_arr) == True))  # noqa: E712
 
             # Perform bilinear interpolation
             x_valid = xx[nan_neighbour_arr]

--- a/icenet/model/train.py
+++ b/icenet/model/train.py
@@ -3,11 +3,8 @@ import datetime as dt
 import json
 import logging
 import os
-import pkg_resources
 import random
 import time
-
-from pprint import pformat
 
 import numpy as np
 import pandas as pd

--- a/icenet/plotting/data.py
+++ b/icenet/plotting/data.py
@@ -14,7 +14,6 @@ from icenet.data.cli import date_arg
 from icenet.data.dataset import IceNetDataSet
 from icenet.utils import setup_logging
 
-import matplotlib as mpl
 import matplotlib.pyplot as plt
 from mpl_toolkits.axes_grid1 import make_axes_locatable
 

--- a/icenet/plotting/forecast.py
+++ b/icenet/plotting/forecast.py
@@ -737,9 +737,9 @@ def standard_deviation_heatmap(metric: str,
     if metric in ["mae", "mse", "rmse"]:
         ylabel = f"SIC {metric.upper()} (%)"
     elif metric == "binacc":
-        ylabel = f"Binary accuracy (%)"
+        ylabel = "Binary accuracy (%)"
     elif metric == "sie":
-        ylabel = f"SIE error (km)"
+        ylabel = "SIE error (km)"
         
     # plot heatmap of standard deviation for IceNet
     fig, ax = plt.subplots(figsize=(12, 6))
@@ -903,9 +903,9 @@ def plot_metrics_leadtime_avg(metric: str,
     if metric in ["mae", "mse", "rmse"]:
         ylabel = f"SIC {metric.upper()} (%)"
     elif metric == "binacc":
-        ylabel = f"Binary accuracy (%)"
+        ylabel = "Binary accuracy (%)"
     elif metric == "sie":
-        ylabel = f"SIE error (km)"
+        ylabel = "SIE error (km)"
     
     if average_over == "all":
         # averaging metric over leadtime for all forecasts

--- a/icenet/plotting/utils.py
+++ b/icenet/plotting/utils.py
@@ -6,6 +6,7 @@ import re
 
 import cartopy.crs as ccrs
 import matplotlib.pyplot as plt
+import numpy as np
 import pandas as pd
 import xarray as xr
 

